lib/manageiq/api/common/request.rb

```diff
@@ -31,17 +44,24 @@ def self.with_request(request)
         end
 
         def self.current_forwardable
-          raise ManageIQ::API::Common::HeadersNotSet, "Current headers have not been set" unless current
-          current.forwardable
+          current!.forwardable
         end
 
         attr_reader :headers, :original_url
 
-        def initialize(headers:, original_url:, **kwargs)
+        def initialize(headers:, original_url:, **_kwargs)
           headers = from_hash(headers) if headers.kind_of?(Hash)
           @headers, @original_url = headers, original_url
         end
 
+        def request_id
+          headers[REQUEST_ID_KEY]
+        end
+
+        def identity
+          JSON.parse(Base64.decode64(headers[IDENTITY_KEY]))
+        end
+
         def user
           @user ||= User.new
```

Comment:
Can we return `nil` if header does not have `IDENTITY_KEY`? The other option is to raise an error.

Reply:
I agree this code should do that as well (or perhaps User.new should do that and it would bubble up to here).